cache/in-memory/src/event/guild.rs

```diff
@@ -0,0 +1,400 @@
+use crate::{
+    config::ResourceType,
+    model::{CachedGuild, CachedPresence},
+    InMemoryCache, UpdateCache,
+};
+use dashmap::DashMap;
+use std::{collections::HashSet, hash::Hash};
+use twilight_model::{
+    gateway::payload::{GuildCreate, GuildDelete, GuildUpdate},
+    guild::Guild,
+    id::GuildId,
+};
+
+impl InMemoryCache {
+    /// Gets a guild by ID.
+    ///
+    /// This is an O(1) operation. This requires the [`GUILDS`] intent.
+    ///
+    /// [`GUILDS`]: ::twilight_model::gateway::Intents::GUILDS
+    pub fn guild(&self, guild_id: GuildId) -> Option<CachedGuild> {
+        self.0.guilds.get(&guild_id).map(|r| r.clone())
+    }
+
+    fn cache_guild(&self, guild: Guild) {
+        // The map and set creation needs to occur first, so caching states and
+        // objects always has a place to put them.
+        if self.wants(ResourceType::CHANNEL) {
+            self.0.guild_channels.insert(guild.id, HashSet::new());
```

Comment:
Nit for all these: What about `HashSet::with_capacity` since we have lengths right here too?